engine/src/main/java/org/terasology/world/internal/EntityAwareWorldProvider.java

```diff
@@ -377,7 +348,7 @@ private void updateBlockEntityComponents(EntityRef blockEntity, Block oldType, B
 
     private EntityRef createBlockEntity(Vector3ic blockPosition, Block block) {
         EntityBuilder builder = entityManager.newBuilder(block.getPrefab().orElse(null));
-        builder.addComponent(new LocationComponent(new org.joml.Vector3f(blockPosition)));
+        builder.addComponent(new LocationComponent(new Vector3f(blockPosition)));
```

Comment:
seems like we could use a LocationComponent(Vector3ic), but maybe that's out-of-scope for this PR